tracer/src/Datadog.Trace/DatabaseMonitoring/DatabaseMonitoringPropagator.cs

```diff
@@ -73,5 +82,74 @@ internal static string PropagateSpanData(DbmPropagationLevel propagationStyle, s
 
             return string.Empty;
         }
+
+        /// <summary>
+        /// Uses a sql instruction to set a context for the current connection, bearing the span ID and trace ID.
+        /// This is meant to circumvent cache invalidation issues that occur when those values are injected in comment.
+        /// Currently only working for MSSQL (uses an instruction that is specific to it)
+        /// </summary>
+        /// <returns>True if the traceparent information was set</returns>
+        internal static bool PropagateDataViaContext(DbmPropagationLevel propagationLevel, IntegrationId integrationId, IDbConnection? connection, Span span)
+        {
+            if (propagationLevel != DbmPropagationLevel.Full || integrationId != IntegrationId.SqlClient || connection == null)
+            {
+                return false;
+            }
+
+            var stopwatch = System.Diagnostics.Stopwatch.StartNew();
+
+            byte version = 0; // version can have a maximum value of 15 in the current format
+            var sampled = SamplingPriorityValues.IsKeep(span.Context.GetOrMakeSamplingDecision() ?? SamplingPriorityValues.Default);
+            var contextValue = BuildContextValue(version, sampled, span.SpanId, span.TraceId128);
+
+            using (var injectionCommand = connection.CreateCommand())
+            {
+                injectionCommand.CommandText = SetContextCommand;
+
+                var parameter = injectionCommand.CreateParameter();
+                parameter.ParameterName = ContextInfoParameterName;
+                parameter.Value = contextValue;
+                parameter.DbType = DbType.Binary;
+                injectionCommand.Parameters.Add(parameter);
+
+                injectionCommand.ExecuteNonQuery();
+
+                if (Log.IsEnabled(LogEventLevel.Debug))
+                {
+                    // avoid building the string representation in the general case where debug is disabled
+                    Log.Debug("Span data for DBM propagated for {Integration} via context_info with value {ContextValue} (propagation level: {PropagationLevel}", integrationId, HexConverter.ToString(contextValue), propagationLevel);
```

Comment:
`HexConverter` is an internal .NET type that we "vendored" into the library. We have a `HexString` wrapper that we try to use when possible since some of the `HexConverter` APIs are not very friendly. No so much in this case, but it would be nice to be consistent.

Also note `HexString` defaults to lower-case while `HexConverter` defaults to upper-case, in case that matters.

```suggestion
                    Log.Debug("Span data for DBM propagated for {Integration} via context_info with value {ContextValue} (propagation level: {PropagationLevel}", integrationId, HexString.ToHexString(contextValue, lowerCase: false), propagationLevel);
```

Reply:
ok, well this is just of a debug log, I don't think the casing is very important

Reply:
Then you can use the default value: `HexString.ToHexString(contextValue)`. The main point was more about using `HexString` instead of `HexConverter` :)